core/domain/role_services.py

```diff
@@ -154,33 +143,23 @@
 #   Instructions-for-editing-roles-or-actions) before making any changes to
 #   this dict.
 ROLE_ACTIONS = {
-    feconf.ROLE_ID_ADMIN: [],
-    feconf.ROLE_ID_BANNED_USER: [],
+    feconf.ROLE_ID_ADMIN: [
+    ],
+    feconf.ROLE_ID_BANNED_USER: [
+    ],
     feconf.ROLE_ID_COLLECTION_EDITOR: [
-        ACTION_CREATE_COLLECTION,
-        ACTION_EDIT_COLLECTION,
-        ACTION_EDIT_COLLECTION_RIGHTS,
-        ACTION_VIEW_COLLECTION_RIGHTS,
     ],
     feconf.ROLE_ID_EXPLORATION_EDITOR: [
-        ACTION_CREATE_EXPLORATION,
-        ACTION_EDIT_EXPLORATION,
-        ACTION_START_FEEDBACK_THREAD,
-        ACTION_SUGGEST_EDIT_TO_EXPLORATION
     ],
     feconf.ROLE_ID_GUEST: [
-        ACTION_DOWNLOAD_EXPLORATION,
-        ACTION_PLAY_COLLECTION,
-        ACTION_PLAY_EXPLORATION,
-        ACTION_VIEW_EXPLORATION_STATS
+        ACTION_PLAY_PUBLIC_COLLECTION,
+        ACTION_PLAY_PUBLIC_EXPLORATION,
     ],
     feconf.ROLE_ID_MODERATOR: [
-        ACTION_ACCESS_MODERATOR_PAGE,
-        ACTION_UPDATE_FEATURED_ACTIVITIES
+        ACTION_PLAY_PRIVATE_COLLECTION,
```

Comment:
Perhaps this should be PLAY_ANY_PRIVATE_COLLECTION. It was unclear to me for a while whether this referred to the user's own private collections or something more general. If you make this change I recommend adding "_ANY_" to the others as well.